src/linux/osd.c

```diff
@@ -257,3 +257,49 @@ size_t ofi_ifaddr_get_speed(struct ifaddrs *ifa)
 }
 
 #endif /* HAVE_ETHTOOL */
+
+int ofi_ifaddr_get_mtu(const struct ifaddrs *ifa)
+{
+	FILE *fd;
+	char *line = NULL;
+	size_t len = 0;
+	char *mtu_filename_prefix = "/sys/class/net/";
+	char *mtu_filename_suffix = "/mtu";
+	char *mtu_filename;
+	int mtu;
+	/* IF_NAMESIZE includes NULL-terminated symbol */
+	size_t filename_len = strlen(mtu_filename_prefix) +
+				     strlen(mtu_filename_prefix) +
+				     IF_NAMESIZE;
+
+	mtu_filename = calloc(1, filename_len);
+	if (!mtu_filename)
+		return 0;
+
+	snprintf(mtu_filename, filename_len, "%s%s%s",
+		 mtu_filename_prefix, ifa->ifa_name, mtu_filename_suffix);
```

Comment:
Can use `asprintf` instead. That way, you don't need to calculate the size of the string, and thus don't need to define the variable for the prefix and suffix.

Reply:
Thanks for the suggestions. All applied in the update.